code/modules/mob/living/carbon/carbon.dm

```diff
@@ -644,12 +644,23 @@
 /mob/living/carbon/proc/update_tint()
 	if(!GLOB.tinted_weldhelh)
 		return
+
+	var/change_overlay = FALSE
+	if(wear_mask && wear_mask.different_overlay)
+		change_overlay = TRUE
+
 	tinttotal = get_total_tint()
 	if(tinttotal >= TINT_BLIND)
 		become_blind(EYES_COVERED)
 	else if(tinttotal >= TINT_DARKENED)
 		cure_blind(EYES_COVERED)
-		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, 2)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, (change_overlay ? 4 : 2), clamp(tinttotal * 100, 0, 255))
+	else if(tinttotal >= TINT_LIGHTER)
+		cure_blind(EYES_COVERED)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, (change_overlay ? 4 : 1), clamp(tinttotal * 100, 0, 255))
+	else if(tinttotal > TINT_LIGHT)
+		cure_blind(EYES_COVERED)
+		overlay_fullscreen("tint", /atom/movable/screen/fullscreen/impaired, (change_overlay ? 4 : 1), clamp(tinttotal * 100, 0, 255))
```

Comment:
arent these the same exact thing

Reply:
no because the tinttotal var get passed as alpha, so in the first one, you get darker alpha, in the second one, you get it ligher

Reply:
but it passes it anyway?

Reply:
yes but the first one is from 1 to 1.99, the second one is from 0.1 to 0.99
currently the second one should be unused tho since there aren't any clothing with less than 1 tint but i added it for modularity

Reply:
But the actual line `overlay_fullscreen`  does not change based on the the tinttotal value, on the value in tinttotal  changes.

Reply:
ahh yes i see what you mean, in the second one should be `change_overlay ? 3 : 1` instead of `change_overlay ? 4 : 1` i'll fix it